### PR TITLE
[vcpkg] Clean up command switch code

### PR DESCRIPTION
**Pull request by strega-nil**
Make it more similar to the non-command switch code.

### CODE REVIEW

File: toolsrc/src/vcpkg/vcpkgcmdarguments.cpp [hunk 1]
```diff
@@ -407,98 +390,107 @@ namespace vcpkg
...
-                else
-                {
-                    const auto& value = it->second.value_or_exit(VCPKG_LINE_INFO);
-                    if (value.front().empty())
+                    if (v.empty())
                     {
-                        // Fail when not given a value, e.g.: "vcpkg install sqlite3 --additional-ports="
-                        System::printf(
-                            System::Color::error, "Error: The option '%s' must be passed an argument.\n", option.name);
+                        System::printf(System::Color::error,
```
Comment (ras0219) [inline toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:461]: This doesn't handle if the option was accidentally passed as a switch.

File: toolsrc/src/vcpkg/vcpkgcmdarguments.cpp [hunk 2]
```diff
@@ -407,98 +390,107 @@ namespace vcpkg
             }
         }
 
-        auto options_copy = this->optional_command_arguments;
-        for (auto&& option : command_structure.options.switches)
+        auto switches_copy = this->command_switches;
+        auto options_copy = this->command_options;
```
Comment (ras0219) [inline toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:394]: What benefit is gained by splitting these up into two categories? It seems that below we always need to check both sets in order to provide decent error messages which increases code complexity.

Comment (strega-nil) [reply toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:394]: The idea is that it normalizes switches vs options, and means we can add more things in the future (like `--no-dry-run` being just a switch `dry-run` with value `false`)

File: toolsrc/src/vcpkg/vcpkgcmdarguments.cpp (hunk 1 above)
Comment (strega-nil) [reply toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:461]: No, but the loop below does.

File: toolsrc/src/vcpkg/vcpkgcmdarguments.cpp [hunk 3]
```diff
@@ -215,14 +212,10 @@ namespace vcpkg
...
     // returns true if this does parse this argument as this option
-    // REQUIRES: Strings::starts_with(argument, "--");
     template<class T>
-    static bool try_parse_argument_as_switch(StringView option, StringView argument, T& place)
+    static bool try_parse_argument_as_switch(StringView option, StringView arg, T& place)
     {
-        // remove the first two '-'s
-        auto arg = argument.substr(2);
-
-        if (equals_modulo_experimental(arg, option))
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:225]: Did handling `x-` get dropped here?

Comment (strega-nil) [reply toolsrc/src/vcpkg/vcpkgcmdarguments.cpp:225]: Yeah

**Review by ras0219-msft — APPROVED** [review]
It would be worth merging from master and performing one final sweep through all commands to ensure every argument has had the `--` prefix removed.

**Pull request merged**